pkg/processor/worker.go

```diff
@@ -309,31 +309,41 @@ func (wrk *worker) createResource(resClient *dynamic.ResourceClient, spec *unstr
 	return nil, true, err
 }
 
-func (wrk *worker) updateResource(resClient *dynamic.ResourceClient, spec *unstructured.Unstructured, obj runtime.Object) (resUpdated *unstructured.Unstructured, retriableError bool, e error) {
-	response, ok := obj.(*unstructured.Unstructured)
+func (wrk *worker) updateResource(bundle *smith.Bundle, resClient *dynamic.ResourceClient, spec *unstructured.Unstructured, obj runtime.Object) (resUpdated *unstructured.Unstructured, retriableError bool, e error) {
+	existingObj, ok := obj.(*unstructured.Unstructured)
 	if !ok {
-		response = &unstructured.Unstructured{
+		existingObj = &unstructured.Unstructured{
 			Object: make(map[string]interface{}),
 		}
-		if err := converter.ToUnstructured(obj, &response.Object); err != nil {
+		if err := converter.ToUnstructured(obj, &existingObj.Object); err != nil {
 			// Unexpected error
 			return nil, false, err
 		}
 	}
 
+	// Check that the object is not marked for deletion
+	if existingObj.GetDeletionTimestamp() != nil {
+		return nil, false, fmt.Errorf("object %v %q is marked for deletion", existingObj.GroupVersionKind(), existingObj.GetName())
+	}
+
+	// Check that this bundle owns the object
+	if !isOwner(existingObj, bundle) {
```

Comment:
So the proposed behaviour here is to NOT do adoption - i.e. if controller reference is missing or not pointing to the bundle it is not updated/added and an error is returned instead.
There is a small issue here - when an orphan that was taking up space and preventing bundle from progressing is deleted, nothing pokes the bundle so that it can retry to proceed. Need to think about it...